routers/repo/issue.go

```diff
@@ -720,11 +720,16 @@ func ViewIssue(ctx *context.Context) {
 func GetActionIssue(ctx *context.Context) *models.Issue {
 	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
 	if err != nil {
-		if models.IsErrIssueNotExist(err) {
-			ctx.Error(404, "GetIssueByIndex")
-		} else {
-			ctx.Handle(500, "GetIssueByIndex", err)
-		}
+		ctx.NotFoundOrServerError("GetIssueByIndex", models.IsErrIssueNotExist, err)
+		return nil
+	}
+	if issue.IsPull && !ctx.Repo.Repository.UnitEnabled(models.UnitTypePullRequests) ||
+		!issue.IsPull && !ctx.Repo.Repository.UnitEnabled(models.UnitTypeIssues) {
+		ctx.Handle(404, "UnitEnabled", nil)
```

Comment:
Error message should be UnitDisabled

Reply:
@daviian no, there should be function name that returned error so that from log entries it would be possible to understand what happened

Reply:
From that point of view yes, but it is confusing at the first moment.

Reply:
That is how it is used in rest of the code :)

Reply:
@lafriks However in this case the UnitEnabled doesn't even return an error and 404 happens if the required unit is not enabled.